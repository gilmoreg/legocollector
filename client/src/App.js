import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Login from './components/Login/Login';
import { login, logout } from './state/actions';
import './App.css';

const checkProfile = () => {
  const profile = JSON.parse(
    window.localStorage.getItem('legocollectorProfile'));
  console.log(profile);
  if (profile && profile.email && profile.token) return profile;
  return null;
};

class App extends Component {
  componentWillMount() {
    // Check for stored profile and set login state accordingly
    const profile = checkProfile();
    if (profile) this.props.dispatch(login(profile));
    else this.props.dispatch(logout);
  }

  render() {
    return (
      <div className="App">
        { this.props.loggedIn ?
          <span>Logged in as {this.props.email}</span> :
          <Login /> }
      </div>
    );
  }
}

App.propTypes = {
  dispatch: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool,
  email: PropTypes.string,
};

App.defaultProps = {
  loggedIn: false,
  email: '',
};

const mapStateToProps = state => ({
  loggedIn: state.loggedIn,
  email: state.email,
});

export default connect(mapStateToProps)(App);
