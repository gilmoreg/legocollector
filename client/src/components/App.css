.App {
  text-align: center;
}

.App .WatchView {
  margin-bottom: 7rem;
}