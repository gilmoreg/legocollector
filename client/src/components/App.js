import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { login, logout, fetchWatches } from '../state/actions';
import { API_URL } from '../config';
import Header from './Header/Header';
import Welcome from './Welcome/Welcome';
import Watch from './Watch/Watch';
import AddWatchButton from './AddWatch/AddWatchButton';
import AddWatchModal from './AddWatch/AddWatchModal';
import Instructions from './Welcome/Instructions';
import './App.css';

const checkProfile = () => {
  const profile = JSON.parse(
    window.localStorage.getItem('legocollectorProfile'));
  if (profile && profile.email && profile.token) return profile;
  return null;
};

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      newWatchModalOpen: false,
      serverAwake: false,
      serverError: '',
    };
  }

  componentDidMount() {
    // Check for stored profile and set login state accordingly
    const profile = checkProfile();
    if (profile) {
      this.props.dispatch(login(profile));
      this.props.dispatch(fetchWatches(profile.token));
    } else this.props.dispatch(logout);
    // Keep the user notified of progress waking the server up
    fetch(`${API_URL}/`)
      .then(() => this.setState({ serverAwake: true }))
      .catch(() => this.setState({ serverError: 'Sorry, we couldn\'t contact the server.' }));
  }

  render() {
    const watches = this.props.watches.map(watch =>
      <Watch watch={watch} key={watch.id} />);

    return (
      <div className="App">
        <Header loggedIn={this.props.loggedIn} email={this.props.email} />
        { this.props.loggedIn ?
          <div className="WatchView">
            { this.state.serverAwake ? '' : <h2 data-text="Please wait while we gather your sets...">Please wait while we gather your sets...</h2>}
            { this.state.serverError ? <h2>{this.state.serverError}</h2> : ''}
            { watches.length ? watches : <Instructions /> }
          </div>
          :
          <Welcome />
        }
        { this.props.loggedIn ?
          <AddWatchButton
            openModal={() => this.setState({ newWatchModalOpen: !this.state.newWatchModalOpen })}
          /> :
          ''
        }
        <AddWatchModal
          open={this.state.newWatchModalOpen}
          close={() => this.setState({ newWatchModalOpen: false })}
        />
      </div>
    );
  }
}

App.propTypes = {
  dispatch: PropTypes.func.isRequired,
  loggedIn: PropTypes.bool,
  email: PropTypes.string,
  watches: PropTypes.array,
  className: PropTypes.string,
};

App.defaultProps = {
  loggedIn: false,
  email: '',
  watches: [],
};

const mapStateToProps = state => ({
  loggedIn: state.loggedIn,
  email: state.email,
  watches: state.watches,
});

export default connect(mapStateToProps)(App);
