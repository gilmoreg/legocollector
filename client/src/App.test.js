import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { App } from './App';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  window.localStorage = {
    getItem: () => JSON.stringify({
      email: 'test@test.com',
      token: '123',
    }),
  }

  const wrapper = shallow(
    <App dispatch={()=>{}} />
  );
  expect(wrapper).toMatchSnapshot();
  window.localStorage = undefined;
});
