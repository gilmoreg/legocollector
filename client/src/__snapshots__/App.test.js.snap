// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dispatch={[Function]}
    email=""
    loggedIn={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Login) />,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Login) />,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
